airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -318,15 +322,23 @@ def _auth_azure(self, _client: hvac.Client) -> None:
             )
 
     def _auth_aws_iam(self, _client: hvac.Client) -> None:
-        if self.auth_mount_point:
+        if self.key_id or self.secret_id or self.role_id:
             _client.auth.aws.iam_login(
                 access_key=self.key_id,
                 secret_key=self.secret_id,
                 role=self.role_id,
                 mount_point=self.auth_mount_point,
             )
-        else:
-            _client.auth.aws.iam_login(access_key=self.key_id, secret_key=self.secret_id, role=self.role_id)
+        elif self.aws_conn_id:
```

Comment:
@pankajastro - we will likely want to follow the method [here](https://github.com/apache/airflow/blob/providers-amazon/8.19.0/airflow/providers/amazon/aws/secrets/secrets_manager.py#L180-L198), as I don't believe we can use a connection. The details would be from the backend kwargs.

Reply:
I believe users could create an AWS connection and then pass the connection ID in the backend kwargs. This approach would allow users to manage fewer parameters in the configuration or environment variables. Additionally, if users are directly using secret key/access key, they can utilize the connection, ensuring their credentials are protected in the metadata database. wdyt or I'm missing something?

Reply:
Thanks Pankaj. My understanding is that it is the standard is to configure and manage all Secrets Backend configurations in `backend_kwargs`.

Reply:
Removed aws_conn and added `arn_role` param PTAL. example config
```
AIRFLOW__SECRETS__BACKEND_KWARGS='{"kv_engine_version": 1, "mount_point": "kv", "url": "http://35.161.42.91:8200/", "auth_type": "aws_iam", "arn_role": "arn:aws:iam::<AWS_ACCOUNT>:role/role_name}'
```